### PR TITLE
nmap: update examples to be more practical and useful

### PR DESCRIPTION
**Pull request by fazlearefin**
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/nmap.md [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+
+`nmap -p {{port1,port2,…,portN}} {{address_or_addresses}}`
 
-`nmap -T4 {{address_or_addresses}}`
+- Perform service and version detection of the top 1000 ports using default NSE scripts; writing results ('-oN') to output file:
 
-- Scan a specific list of ports (use `-p-` for all ports `1-65535`):
+`nmap -sC -sV -oN {{top-1000-ports.txt}} {{address_or_addresses}}`
 
-`nmap -p {{port1,port2,…,portN}} {{address_or_addresses}}`
```
Comment (CleanMachine1) [inline pages/common/nmap.md:29]: I'd argue that this is a useful command to have

Comment (CleanMachine1) [reply pages/common/nmap.md:29]: ^ the previous one

Comment (fazlearefin) [reply pages/common/nmap.md:29]: @CleanMachine1 I am bit confused, you mean this one is useful:

```
nmap -T4 {{address_or_addresses}}
```

File: pages/common/nmap.md [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-`nmap {{ip_or_hostname}} {{optional_another_address}}`
-
 - Also enable scripts, service detection, OS fingerprinting and traceroute:
 
 `nmap -A {{address_or_addresses}}`
 
-- Assume good network connection and speed up execution:
+- Scan a specific list of ports (use '-p-' for all ports from 1 to 65535):
+
+`nmap -p {{port1,port2,…,portN}} {{address_or_addresses}}`
```
Comment (sbrl) [inline pages/common/nmap.md:21]: The `…` should be an ascii `...` here.

File: pages/common/nmap.md [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
+- Scan web server running on standard ports 80 and 443 using all available 'http-*' NSE scripts:
 
-`nmap -sSU {{address_or_addresses}}`
+`nmap --script "http-*" {{address_or_addresses}} -p 80,443`
 
-- Perform full port, service, version detection scan with all default NSE scripts active against a host to determine weaknesses and info:
+- Perform a stealthy very slow scan ('-T0') trying to avoid detection by IDS/IPS and use decoy ('-D') source IP addresses:
 
-`nmap -sC -sV {{address_or_addresses}}`
+`nmap -T0 -D {{decoy1_ipaddress,decoy2_ipaddress,…,decoyN_ipaddress}} {{address_or_addresses}}`
```
Comment (sbrl) [inline pages/common/nmap.md:37]: Same here. `…` → `...`

File: pages/common/nmap.md (hunk 1 above)
Comment (sbrl) [reply pages/common/nmap.md:29]: I believe @CleanMachine1 means this example:
```markdown
- Perform service and version detection of the top 1000 ports using default NSE scripts; writing results ('-oN') to output file:

`nmap -sC -sV -oN {{top-1000-ports.txt}} {{address_or_addresses}}`
```

**Review by sbrl — COMMENTED** [review]
Thanks for the new page! I've left some comments below for you to review.

File: pages/common/nmap.md (hunk 3 above)
Comment (fazlearefin) [reply pages/common/nmap.md]: Fixed

File: pages/common/nmap.md (hunk 2 above)
Comment (fazlearefin) [reply pages/common/nmap.md]: Fixed

File: pages/common/nmap.md (hunk 1 above)
Comment (fazlearefin) [reply pages/common/nmap.md]: `nmap -sC -sV -oN {{top-1000-ports.txt}} {{address_or_addresses}}` is a very useful command used by most professional pentesters and CTF players. Please watch one of ippsec's hacking a box demo here https://www.youtube.com/watch?v=4e4wKDrANog 

(IppSec is a well respected individual in the inforsec space)

**Review by sbrl — APPROVED** [review]

**Pull request merged**